FirebaseFunctions/Sources/Callable+Codable.swift

```diff
@@ -160,3 +165,183 @@ public struct Callable<Request: Encodable, Response: Decodable> {
+      }
+
+      switch onlyKey {
+      case .message:
+        self = try Self
+          .message(container.decode(Message.self, forKey: .message))
+      case .result:
+        self = try Self
+          .result(container.decode(Result.self, forKey: .result))
+      }
+    } catch {
+      throw FunctionsError(.dataLoss, userInfo: [NSUnderlyingErrorKey: error])
+    }
+  }
+}
+
+@available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+public extension Callable {
+  /// Creates a stream that yields responses from the streaming callable function.
+  ///
+  /// The request to the Cloud Functions backend made by this method
+  /// automatically includes a FCM token to identify the app instance.
+  /// If a user is logged in with Firebase Auth, an auth ID token for the
+  /// user is included. If App Check is integrated, an app check token is
+  /// included.
+  ///
+  /// Firebase Cloud Messaging sends data to the Firebase backend
+  /// periodically to collect information regarding the app instance. To
+  /// stop this, see `Messaging.deleteData()`. It resumes with a new FCM
+  /// Token the next time you call this method.
+  ///
+  /// - Important: The final result returned by the callable function is only
+  ///   accessible when using `StreamResponse` as the `Response` generic
+  ///   type.
+  ///
+  /// Example of using `stream` _without_ `StreamResponse`:
+  /// ```swift
+  /// let callable: Callable<MyRequest, MyResponse> = // ...
+  /// let request: MyRequest = // ...
+  /// let stream = try callable.stream(request)
+  /// for try await response in stream {
+  ///   // Process each `MyResponse` message
+  ///   print(response)
+  /// }
+  /// ```
+  ///
+  /// Example of using `stream` _with_ `StreamResponse`:
+  /// ```swift
+  /// let callable: Callable<MyRequest, StreamResponse<MyMessage,
+  /// MyResult>> = // ...
+  /// let request: MyRequest = // ...
+  /// let stream = try callable.stream(request)
+  /// for try await response in stream {
+  ///   switch response {
+  ///   case .message(let message):
+  ///     // Process each `MyMessage`
+  ///     print(message)
+  ///   case .result(let result):
+  ///     // Process the final `MyResult`
+  ///     print(result)
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// - Parameter data: The `Request` data to pass to the callable function.
+  /// - Throws: A ``FunctionsError`` if the parameter `data` cannot be
+  ///   encoded.
+  /// - Returns: A stream wrapping responses yielded by the streaming callable
+  ///   function or a ``FunctionsError`` if an error occurred.
+  func stream(_ data: Request? = nil) throws -> AsyncThrowingStream<Response, Error> {
+    let encoded: Any
+    do {
+      encoded = try encoder.encode(data)
+    } catch {
+      throw FunctionsError(.invalidArgument, userInfo: [NSUnderlyingErrorKey: error])
+    }
+
+    return AsyncThrowingStream { continuation in
+      Task {
+        do {
+          for try await response in callable.stream(encoded) {
```

Comment:
Does this code emit any warnings in Swift 6? Although it would be nice to avoid the unstructured concurrency, potentially this approach could work in Vertex AI too: https://github.com/firebase/firebase-ios-sdk/blob/ea8fb07f37a3dc647bb494a7a6b17b9eb6f78c26/FirebaseVertexAI/Sources/GenerativeModel.swift#L188-L190

Reply:
Ah, too bad! Thanks for letting me know though.

Reply:
I'm unable to address all of the sendability issues right now (self needs to be Sendable), but we can get ahead of the two red errors in the above screenshot. The `Response: any Decodable` generic arg should also be Sendable. 

This can be done by enabling this new stream API when `Response: any Decodable && Sendable`.

I also added the constraint for the `Request` arg, but I'm less sure that matters... My rationale is that it's easier to remove the Sendable constraint than add it (breaking change).

https://github.com/firebase/firebase-ios-sdk/pull/14395/commits/ec33fa86708ff9344871f16046765a2ed5c9afbb